### ISSUE TITLE
Why this project not implemented asynchronous

### ISSUE THREAD

**Issue by Turall** [tiangolo/full-stack-fastapi-postgresql#354]
Fastapi based on starlette and starlette based on asyncio. Why was this project implemented using synchronous programming? I just want to understand the reason. Thanks in advance

**Comment by haviduck** [comment]
as far as i understand and i certainly cant speak for tiangolo, this is a boilerplate stack that gives you a great starting point to learn and build your app.  it hasnt been updated for a while and it has a few bugs. besides that we are meant to learn from it and convert it into our own.   adding async defs is well documented and is just a matter of adding your own code. its pretty awesome that they bother giving us a freebie like this.

is there any async issues i can help you with perhaps?

**Comment by Turall** [comment]
> as far as i understand and i certainly cant speak for tiangolo, this is a boilerplate stack that gives you a great starting point to learn and build your app. it hasnt been updated for a while and it has a few bugs. besides that we are meant to learn from it and convert it into our own. adding async defs is well documented and is just a matter of adding your own code. its pretty awesome that they bother giving us a freebie like this.
> 
> is there any async issues i can help you with perhaps?

Thanks for the answer. Yes, this is a great project and thanks to everyone who worked on it. I don't have  problem with async, I just wanted to understand why they didn't choose the asynchronous approach.

**Comment by pramadito** [comment]
here's why:

> you can mix def and async def in your path operation functions as much as you need and define each one using the best option for you. FastAPI will do the right thing with them.
> 
> Anyway, in any of the cases above, FastAPI will still work asynchronously and be extremely fast.
> 
> But by following the steps above, it will be able to do some performance optimizations.

[source](https://fastapi.tiangolo.com/async/#in-a-hurry)

tl;dr no matter using sync or async in fastapi, fastapi will automatically configurate it to async.

**Comment by Turall** [comment]
Thank you for your response and support.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @Turall  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
